editorconfig_plugin/gedit2.py

```diff
@@ -39,6 +40,6 @@ def get_document_properties(self, document):
         """Call EditorConfig core and return properties dict for document"""
         if document:
             file_uri = document.get_uri()
-            if file_uri and file_uri.startswith("file:///"):
-                return self.get_properties_from_filename(file_uri[7:])
+            if file_uri and file_uri.startswith('file://'):
+                return self.get_properties_from_filename(unquote(urlparse(file_uri).path))
```

Comment:
I'll have to track down GEdit 2 API documentation, to figure out if this one can use similar calls. Though it's debatable whether providing a gedit2 plugin really makes any sense anymore.

Reply:
Hmm. It seems that while `document.get_uri()` returned a `gchar` string containing the URI, there was also a  `document.get_location()` which, similar to GEdit 3, returned an ancient-history version of a `GFile`. ([ref](https://gitlab.gnome.org/GNOME/gedit/-/blob/7013f7e0330107d3106dfc98fafab15da82e0789/gedit/gedit-document.h))

What did the `GFile` API look like in 2010, tho? Absolutely no idea.

Reply:
...Huh, I guess even back in 2010 `GFile` was provided by Gio, which means its API should be fairly similar to what it is today. GLib has evolved _very_ little over the years. And neither `has_uri_scheme()` nor `get_path()` is shown as being a later addition to the `GFile` API, so they should exist just the same even in GEdit 2. 

Looks like this one can be written the same as in GEdit 3, just using `document.get_location()` instead of `document.get_file().get_location()`.